src/LogIndexOperationsTrait.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\elasticsearch;
+
+use oat\tao\model\search\index\IndexDocument;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+/**
+ * @internal To be used only by lib-tao-elasticsearch
+ */
+trait LogIndexOperationsTrait
+{
+    private function logCompletion(LoggerInterface $logger, int $count, int $visited, int $skipped, int $exceptions): void
+    {
+        if (($count != $visited) || ($exceptions > 0) || ($skipped > 0)) {
+            $logger->warning(
+                sprintf(
+                    "%d / %d items were processed (%d skipped, %d exceptions)",
+                    $count,
+                    $visited,
+                    $skipped,
+                    $exceptions
+                )
+            );
+        } else {
+            $logger->debug(
+                sprintf(
+                    "Processed %d items (no exceptions, no skipped items)",
+                    $count,
+                    $skipped
+                )
+            );
+        }
+    }
+
+    private function logAddingDocumentToQueue(
+        LoggerInterface $logger,
+        IndexDocument $document,
+        string $indexName
+    ): void
+    {
+        $logger->info(
+            ($document ? sprintf('[documentId: "%s"] ', $document->getId()) : '') .
+            sprintf(
```

Comment:
**(Optional)** [PSR-2](https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md#multi-line-declarationconditionconcatenation): `All operators should go in the newline as first character`

```suggestion
            ($document ? sprintf('[documentId: "%s"] ', $document->getId()) : '')
            . sprintf(
```